theano/gof/cmodule.py

```diff
@@ -640,480 +653,427 @@ def refresh(self, age_thresh_use=None, delete_if_problem=False):
+                                        key_pkl)
+                    rmtree(root, ignore_nocleanup=True,
+                           msg="missing module file", level=logging.INFO)
+                continue
+                if (time_now - last_access_time(entry)) < age_thresh_use:
+                    _logger.debug('refresh adding %s', key_pkl)
+
+                    def unpickle_failure():
+                        _logger.info("ModuleCache.refresh() Failed to "
+                                     "unpickle cache file %s", key_pkl)
+
                     try:
-                        entry = module_name_from_dir(root, files=files)
-                    except ValueError:  # there is a key but no dll!
-                        if not root.startswith("/tmp"):
-                            # Under /tmp, file are removed periodically by the
-                            # os. So it is normal that this happens from time
-                            # to time.
-                            _logger.warning("ModuleCache.refresh() Found key "
-                                            "without dll in cache, deleting it. %s",
-                                            key_pkl)
-                        _rmtree(root, ignore_nocleanup=True,
-                                msg="missing module file", level=logging.INFO)
+                        with open(key_pkl, 'rb') as f:
+                            key_data = cPickle.load(f)
+                    except EOFError:
+                        # Happened once... not sure why (would be worth
+                        # investigating if it ever happens again).
+                        unpickle_failure()
+                        rmtree(root, ignore_nocleanup=True,
+                               msg='broken cache directory [EOF]',
+                               level=logging.WARNING)
+                        continue
+                    except ValueError:
+                        # This can happen when we have bad config value
+                        # in the cuda.nvcc_compiler.py file.
+                        # We should not hide it here, as this will cause
+                        # an unrelated error to appear.
+                        raise
+                    except Exception:
+                        unpickle_failure()
+                        if delete_if_problem:
+                            rmtree(root, ignore_nocleanup=True,
+                                   msg='broken cache directory',
+                                   level=logging.INFO)
+                        else:
+                            # This exception is often triggered by keys
+                            # that contain references to classes that have
+                            # not yet been imported (e.g. when running two
+                            # different Theano-based scripts). They are not
+                            # necessarily broken, but we cannot load them
+                            # here.
```

Comment:
Bad old comment (not in your PR). Can you change the last work to "now" instead of "here". And add that they will be reloaded if they are needed.